### PR TITLE
Free a couple of memory leaks in the SOF plugin

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: tools/plugin/alsaplug/tplg.c [hunk 1]
```diff
@@ -1456,7 +1456,12 @@ void plug_free_topology(snd_sof_plug_t *plug)
 
 	list_for_item_safe(item, _item, &plug->widget_list) {
 		struct tplg_comp_info *comp_info = container_of(item, struct tplg_comp_info, item);
+		struct sof_ipc4_available_audio_format *available_fmts = &comp_info->available_fmt;
 
+		if (available_fmts->output_pin_fmts)
+			free(available_fmts->output_pin_fmts);
+		if (available_fmts->input_pin_fmts)
+			free(available_fmts->input_pin_fmts);
```
Comment (kv2019i) [inline tools/plugin/alsaplug/tplg.c:1465]: Isn't this if-check redundannt, NULL is ok to free

**Review by kv2019i — COMMENTED** [review]
One comment to first patch, otherwise looks good.

**Review by lgirdwood — COMMENTED** [review]
Ack, LGTM with comment from @kv2019i.
Btw, does this mean valgrind runs with 0 errors ?

File: tools/plugin/alsaplug/tplg.c (hunk 1 above)
Comment (ranj063) [reply tools/plugin/alsaplug/tplg.c]: ack fixed now

**Comment by ranj063** [comment]
> Ack, LGTM with comment from @kv2019i. Btw, does this mean valgrind runs with 0 errors ?

not yet @lgirdwood. Im working to get there

**Review by abonislawski — APPROVED** [review]

File: tools/plugin/alsaplug/tplg.c (hunk 1 above)
Comment (kv2019i) [reply tools/plugin/alsaplug/tplg.c]: @ranj063 the fix went into the wrong patch

**Review by lgirdwood — COMMENTED** [review]
Should these 2 patches be squashed ?

**Comment by lgirdwood** [comment]
@ranj063 ping.

**Comment by lgirdwood** [comment]
@ranj063 ping

**Comment by ranj063** [comment]
> @ranj063 ping

Sorry for the delay @lgirdwood . Fixed now

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
@wszypelt Can you check? This seems to be stuck..

**Comment by kv2019i** [comment]
@wszypelt @ranj063 Still seeing failure. Shall I just proceed with merge? This is touching plugin code that is not even built into the FW, so the quickbuild tests are in no way relevant to this change.

**Comment by wszypelt** [comment]
@kv2019i after checking the tests, one issue appeared, which is sporadic, in my opinion you can merge, but if you still have about an hour, the result should already be green

**Pull request merged**